src/Core/Extensions/GraphQLBuilderCoreExtensions.cs

```diff
@@ -16,12 +17,45 @@ namespace GraphQL.Server
     /// </summary>
     public static class GraphQLBuilderCoreExtensions
     {
+        /// <summary>
+        /// Registers an instance of <see cref="BasicGraphQLExecuter{TSchema}"/> as <see cref="IGraphQLExecuter{TSchema}"/> for
+        /// use with <see cref="GraphQLOptions"/>. It is recommended to use <see cref="IDocumentExecuter"/> directly rather than
+        /// use the <see cref="IGraphQLExecuter{TSchema}"/> interface.
+        /// <br/><br/>
+        /// Also installs <see cref="InstrumentFieldsMiddleware"/> in the schema if specified. (Previous' versions of
+        /// GraphQL.Server would install this middleware automatically.) Specify <see langword="false"/> for <paramref name="installMetricsMiddleware"/>
+        /// if <see cref="GraphQLBuilderExtensions.AddMetrics(DI.IGraphQLBuilder, bool)"/> will be called separately. Note that
+        /// the implementation of <see cref="BasicGraphQLExecuter{TSchema}"/> will set <see cref="ExecutionOptions.EnableMetrics"/>
+        /// if <see cref="GraphQLOptions.EnableMetrics"/> is set.
+        /// </summary>
+        public static DI.IGraphQLBuilder AddServer(this DI.IGraphQLBuilder builder, bool installMetricsMiddleware, Action<GraphQLOptions> configureOptions)
```

Comment:
Regarding all that settings around middleware - too hard. Let's remove `installMetricsMiddleware` parameter and `if (installMetricsMiddleware)
                builder.AddMetrics(false);` from here. One can set in outside that method.

Reply:
Previously the delegate to configure `GraphQLOptions` would previously have been used to set `UseMetrics` to `true`.  With the requested change that will not work.

I think this should be considered with #679 -- see notes I'll add there.

Reply:
(and left as-is in this PR)

Reply:
OK